### PR TITLE
Disble multisite folder creation

### PR DESCRIPTION
**Pull request by djamnazi**
[OO-BCBREAK] Disable multi website in media folder
[OO-DEPRECATED] FolderRepository:findAllRootFolder function in OpenOrchestra\MediaModelBundle\Repository namespace has been deprecated  use findAllRootFolderBySiteId

### CODE REVIEW

File: Media/Model/FolderInterface.php [hunk 1]
```diff
@@ -62,24 +55,12 @@ public function addSubFolder(FolderInterface $subFolder);
...
-    /**
-     * @param array $site
-     */
-    public function addSite(array $site);
-
-    /**
-     * @param array $site
      */
-    public function removeSite(array $site);
+    public function addSiteId($siteId);
```
Comment (alavieille) [inline Media/Model/FolderInterface.php]: rename it setSiteId

File: MediaModelBundle/Document/Folder.php [hunk 2]
```diff
@@ -66,7 +66,7 @@
     public function __construct()
     {
         $this->subFolders = new ArrayCollection();
-        $this->sites = array();
+        $this->siteId = "";
```
Comment (alavieille) [inline MediaModelBundle/Document/Folder.php]: remove this line

File: MediaModelBundle/Document/Folder.php [hunk 3]
```diff
@@ -153,48 +135,18 @@ public function removeSubFolder(FolderInterface $subFolder)
...
-    public function addSite(array $site)
-    {
-        $this->sites[] = $site;
-    }
-
-    /**
-     * @param array $site
      */
-    public function removeSite(array $site)
+    public function addSiteId($siteId)
```
Comment (alavieille) [inline MediaModelBundle/Document/Folder.php]: rename it setSitId

File: MediaModelBundle/Repository/FolderRepository.php [hunk 4]
```diff
@@ -21,9 +21,7 @@ public function findAllRootFolder($siteId = null)
         $qb = $this->createQueryBuilder();
         $qb->field('parent')->equals(null);
         if ($siteId) {
-            $where = "function() { return this.sites && this.sites.length == 0; }";
-            $qb->addOr($qb->expr()->field('sites.siteId')->equals($siteId));
-            $qb->addOr($qb->expr()->field('sites')->where($where));
+            $qb->field('siteId')->equals($siteId);
```
Comment (alavieille) [inline MediaModelBundle/Repository/FolderRepository.php]: i think than findAllRootFolder is deprecated now

**Comment by djamnazi** [comment]
@alavieille PR updated

**Comment by alavieille** [comment]
add tag OO-DEPRECATED

**Comment by djamnazi** [comment]
@alavieille  it's done

**Comment by alavieille** [comment]
can you be more explicit in oo-deprecated tag what is method which replace findAllRootFolder and indicate the class with namespace of this method

**Comment by djamnazi** [comment]
@alavieille it's done

**Comment by itkg-ngilain** [comment]
:+1:

**Pull request merged**